docs/source/data/kedro_dataset_factories.md

```diff
@@ -1,38 +1,51 @@
 # Kedro dataset factories
 You can load multiple datasets with similar configuration using dataset factories, introduced in Kedro `0.18.12`.
 
-The syntax allows you to generalise your configuration and reduce the number of similar catalog entries by matching datasets used in your project's pipelines to dataset factory patterns.
-
-```{warning}
-Datasets are not included in the core Kedro package from Kedro version **`0.19.0`**. Import them from the [`kedro-datasets`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets) package instead.
-From version **`2.0.0`** of `kedro-datasets`, all dataset names have changed to replace the capital letter "S" in "DataSet" with a lower case "s". For example, `CSVDataSet` is now `CSVDataset`.
-```
-
-## How to generalise datasets with similar names and types
-
-Consider the following catalog entries:
+The dataset factories introduce a syntax that allows you to generalise your configuration and reduce the number of similar catalog entries by matching datasets used in your project's pipelines to dataset factory patterns.
 
+For example:
 ```yaml
 factory_data:
   type: pandas.CSVDataset
   filepath: data/01_raw/factory_data.csv
+```
 
-
-process_data:
+With dataset factory, it can be re-written as:
+```yaml
+{name}_data:
   type: pandas.CSVDataset
-  filepath: data/01_raw/process_data.csv
+  filepath: data/01_raw/{name}_data.csv
 ```
 
-The datasets in this catalog can be generalised to the following dataset factory:
+In runtime, the pattern will be matched against the name of the datasets defined in `inputs` or `outputs`.
+```python
+node(
+    func=process_factory,
+    inputs="factory_data",
+    outputs=None,
+),
+...
+```
+
+```{note}
+The factory pattern must always be enclosed in quotes to avoid YAML parsing errors.
+```
 
+Dataset factories is similar to **regular expression** and you can think of it as reversed `f-string`. In this case, the name of dataset `factory_data` matches the pattern `{name}_data` with the `_data` suffix, so it resolves `name` to `factory`.
+
+Similarly, if you have a dataset called `process_data`, it will be resolved as:
 ```yaml
-"{name}_data":
+process_data:
```

Comment:
It was a bit easier for me to understand the example when `factory_data` and `process_data` were mentioned together, as you could easily spot the difference.

Reply:
I have revise it and merge them into one paragraph.